src/main/java/com/patchworkmc/annotation/AnnotationProcessor.java

```diff
@@ -50,24 +59,41 @@ public AnnotationVisitor visitAnnotation(String descriptor, boolean visible) {
 
 			return super.visitAnnotation(descriptor, visible);
 		} else {
-			Patchwork.LOGGER.warn("Unknown class annotation: " + descriptor + " " + visible);
+			annotationStorage.acceptClassAnnotation(descriptor, className);
 			return new AnnotationPrinter(super.visitAnnotation(descriptor, visible));
 		}
 	}
 
 	@Override
 	public FieldVisitor visitField(int access, String name, String descriptor, String signature, Object value) {
-		return new FieldScanner(super.visitField(access, name, descriptor, signature, value));
+		return new FieldScanner(
+				super.visitField(access, name, descriptor, signature, value),
+				annotationStorage,
+				className,
+				name
+		);
```

Comment:
I would prefer if you didn't split constructors / method calls across lines like this. If the line is too long, try to move stuff like `super.visitField(access, name, descriptor, signature, value)` outside of the call first.